sale_product_variant_attribute_tax/tests/test_sale_product_variant_attribute_tax.py

```diff
@@ -57,10 +58,13 @@ def setUpClass(cls):
                 ],
             }
         )
+        cls.product = cls.env["product.product"].create(
+            {"name": "Test template", "product_tmpl_id": cls.product_template.id}
+        )
         order_form = Form(cls.env["sale.order"])
         order_form.partner_id = cls.partner
         with order_form.order_line.new() as order_line_form:
-            order_line_form.product_tmpl_id = cls.product_template
+            order_line_form.product_id = cls.product_template.product_variant_id
```

Comment:
Why this change is necessary?

Reply:
There is a sale_order_line constraint  sale_order_line_accountable_required_fields thats need product_jd completed, when product_tmpl_id is set, product_id is null.